pkg/controller/remotemachineset/remotemachineset_controller_test.go

```diff
@@ -311,6 +314,37 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
 				}
 			}(),
 		},
+		{
+			// Can't control what a user might do in their cluster.
+			name: "Nil ProviderSpec",
+			localExisting: []runtime.Object{
+				testClusterDeployment([]hivev1.MachinePool{
+					testMachinePool("worker", 3, []string{}),
+				}),
+				testSecret(adminKubeconfigSecret, adminKubeconfigSecretKey, testName),
+				testSecret(adminPasswordSecret, adminPasswordSecretKey, testName),
+				testSecret(sshKeySecret, sshKeySecretKey, testName),
+			},
+			remoteExisting: []runtime.Object{
+				testMachineSet("foo-12345-worker-us-east-1a", "worker", true, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1b", "worker", true, 1, 0),
+				func() *machineapi.MachineSet {
+					ms := testMachineSet("foo-12345-worker-us-east-1c", "worker", true, 1, 0)
+					ms.Spec.Template.Spec.ProviderSpec.Value = nil
+					return ms
+				}(),
+			},
+			expectedRemoteMachineSets: func() *machineapi.MachineSetList {
+				return &machineapi.MachineSetList{
+					Items: []machineapi.MachineSet{
+						*testMachineSet("foo-12345-worker-us-east-1a", "worker", true, 1, 0),
+						*testMachineSet("foo-12345-worker-us-east-1b", "worker", true, 1, 0),
+						*testMachineSet("foo-12345-worker-us-east-1c", "worker", true, 1, 0),
+					},
+				}
+			}(),
+			skipAMICheck: true,
```

Comment:
Why not adjust the expected machine set to have a nil provider spec? That is what is expected after all.